src/main/java/org/ballerinax/datamapper/DataMapperStructureVisitor.java

```diff
@@ -46,6 +46,10 @@ public DataMapperStructureVisitor(Map structure) {
 
     @Override
     public void visit(BLangTypeDefinition typeDefinition) {
+        if (!(typeDefinition.typeNode instanceof BLangRecordTypeNode)) {
```

Comment:
As discussed, let's replace these `instanceof` checks with type-tag checks.
We can do that as an improvement, in a separate PR..